### PR TITLE
new update

### PR DESCRIPTION
**Pull request by Gauravbartwal**

### CODE REVIEW

File: src/app/main/login/login.component.ts [hunk 1]
```diff
@@ -27,15 +28,26 @@ export class LoginComponent implements OnInit {
...
   login(){
+    const HTTP_OPTIONS = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/x-www-form-urlencoded',
+      })
+    };
+
+    const body = new HttpParams()
+      .set('username', this.form.value.username)
+      .set('password', this.form.value.password)
```
Comment (plaban1805) [inline src/app/main/login/login.component.ts:39]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
      .set("password", this.form.value.password)
```

File: src/app/main/chat/chat.component.ts [hunk 2]
```diff
@@ -1,4 +1,8 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
```
Comment (plaban1805) [inline src/app/main/chat/chat.component.ts:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
import { Router } from "@angular/router";
```

File: src/app/main/login/login.component.ts [hunk 3]
```diff
@@ -27,15 +28,26 @@
   }
 
   login(){
+    const HTTP_OPTIONS = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/x-www-form-urlencoded',
```
Comment (plaban1805) [inline src/app/main/login/login.component.ts:33]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
        "Content-Type": 'application/x-www-form-urlencoded',
```

File: src/app/main/register/register.component.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import { Router } from '@angular/router';
 import { AuthService } from 'src/app/services/auth.service';
 import {ApiService} from '../../services/api.service'
-
+import { HttpHeaders } from '@angular/common/http';
```
Comment (plaban1805) [inline src/app/main/register/register.component.ts:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
import { HttpHeaders } from "@angular/common/http";
```

File: src/app/main/login/login.component.ts [hunk 5]
```diff
@@ -3,7 +3,8 @@
 import { Router } from '@angular/router';
 import { AuthService } from 'src/app/services/auth.service';
 import {ApiService} from '../../services/api.service'
-
+import { HttpParams } from '@angular/common/http';
+import { HttpHeaders } from '@angular/common/http';
```
Comment (plaban1805) [inline src/app/main/login/login.component.ts:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
import { HttpHeaders } from "@angular/common/http";
```

File: src/app/main/chat/chat.component.ts [hunk 6]
```diff
@@ -1,4 +1,8 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
```
Comment (plaban1805) [inline src/app/main/chat/chat.component.ts:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
import { FormBuilder, FormGroup, Validators } from "@angular/forms";
```

File: src/app/main/login/login.component.ts [hunk 7]
```diff
@@ -27,15 +28,26 @@
...
+    };
+
+    const body = new HttpParams()
+      .set('username', this.form.value.username)
+      .set('password', this.form.value.password)
+      .set('grant_type', 'password');
     let b = this.form.value
     console.log(b)
-    this._api.postTypeRequest('login', b).subscribe((res: any) => {
+    this._api.postTypeRequest('login', body, HTTP_OPTIONS).subscribe((res: any) => {
```
Comment (plaban1805) [inline src/app/main/login/login.component.ts:43]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
    this._api.postTypeRequest("login", body, HTTP_OPTIONS).subscribe((res: any) => {
```

File: src/app/main/register/register.component.ts [hunk 8]
```diff
@@ -29,9 +29,14 @@
...
   register(){
+    const HTTP_OPTIONS = {
+      headers: new HttpHeaders({
+        'Content-Type': 'aplication/json',
+      })
+    };
     let b = this.form.value
     console.log(b)
-    this._api.postTypeRequest('register', b).subscribe((res: any) => {
+    this._api.postTypeRequest('register', b ,HTTP_OPTIONS).subscribe((res: any) => {
```
Comment (plaban1805) [inline src/app/main/register/register.component.ts:39]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
    this._api.postTypeRequest("register", b ,HTTP_OPTIONS).subscribe((res: any) => {
```

File: src/app/main/chat/chat.component.ts [hunk 9]
```diff
@@ -1,4 +1,8 @@
...
+import { Router } from '@angular/router';
+import { AuthService } from 'src/app/services/auth.service';
```
Comment (plaban1805) [inline src/app/main/chat/chat.component.ts:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
import { AuthService } from "src/app/services/auth.service";
```

File: src/app/main/chat/chat.component.ts [hunk 10]
```diff
@@ -7,9 +11,20 @@
...
-  constructor() { }
+  constructor(
+    private _api : ApiService,
+    private _auth: AuthService,
+    private router: Router,
+    public fb: FormBuilder
+  ) { }
 
   ngOnInit(): void {
+    this._api.getTypeRequest('chat').subscribe((res: any) => {
```
Comment (plaban1805) [inline src/app/main/chat/chat.component.ts:22]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
    this._api.getTypeRequest("chat").subscribe((res: any) => {
```

File: src/app/main/chat/chat.component.ts [hunk 11]
```diff
@@ -1,4 +1,8 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { AuthService } from 'src/app/services/auth.service';
+import {ApiService} from '../../services/api.service'
```
Comment (plaban1805) [inline src/app/main/chat/chat.component.ts:5]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
import {ApiService} from "../../services/api.service"
```

File: src/app/main/login/login.component.ts [hunk 12]
```diff
@@ -27,15 +28,26 @@
...
 
   login(){
+    const HTTP_OPTIONS = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/x-www-form-urlencoded',
+      })
+    };
+
+    const body = new HttpParams()
+      .set('username', this.form.value.username)
```
Comment (plaban1805) [inline src/app/main/login/login.component.ts:38]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
      .set("username", this.form.value.username)
```

File: src/app/main/login/login.component.ts [hunk 13]
```diff
@@ -27,15 +28,26 @@
...
+      .set('grant_type', 'password');
     let b = this.form.value
     console.log(b)
-    this._api.postTypeRequest('login', b).subscribe((res: any) => {
+    this._api.postTypeRequest('login', body, HTTP_OPTIONS).subscribe((res: any) => {
       console.log(res)
       if(res.access_token){
         this._auth.setDataInLocalStorage('token', res.access_token)
-        this.router.navigate(['/dashboard'])
+        this.router.navigate(['/chat'])
```
Comment (plaban1805) [inline src/app/main/login/login.component.ts:47]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
        this.router.navigate(["/chat"])
```

File: src/app/main/login/login.component.ts [hunk 14]
```diff
@@ -27,15 +28,26 @@
...
+    const HTTP_OPTIONS = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/x-www-form-urlencoded',
+      })
+    };
+
+    const body = new HttpParams()
+      .set('username', this.form.value.username)
+      .set('password', this.form.value.password)
+      .set('grant_type', 'password');
```
Comment (plaban1805) [inline src/app/main/login/login.component.ts:40]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
      .set("grant_type", 'password');
```

File: src/app/main/login/login.component.ts [hunk 15]
```diff
@@ -3,7 +3,8 @@
 import { Router } from '@angular/router';
 import { AuthService } from 'src/app/services/auth.service';
 import {ApiService} from '../../services/api.service'
-
+import { HttpParams } from '@angular/common/http';
```
Comment (plaban1805) [inline src/app/main/login/login.component.ts:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
import { HttpParams } from "@angular/common/http";
```

File: src/app/main/chat/chat.component.ts [hunk 16]
```diff
@@ -7,9 +11,20 @@
...
+    private _api : ApiService,
+    private _auth: AuthService,
+    private router: Router,
+    public fb: FormBuilder
+  ) { }
 
   ngOnInit(): void {
+    this._api.getTypeRequest('chat').subscribe((res: any) => {
+      console.log(res)
+    }, err => {
```
Comment (plaban1805) [inline src/app/main/chat/chat.component.ts:24]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
    }, (err) => {
```

File: src/app/main/chat/chat.component.ts (hunk 2 above)
Comment (Gauravbartwal) [inline src/app/main/chat/chat.component.ts]: done

**Comment by Gauravbartwal** [comment]
pull request and codacy

**Pull request merged**